x/slashrefund/keeper/withdraw.go

```diff
@@ -16,15 +16,15 @@ func (k Keeper) Withdraw(
 	valAddr sdk.ValAddress,
 	tokens sdk.Coin,
 ) (sdk.Coin, time.Time, error) {
-	
+
 	deposit, found := k.GetDeposit(ctx, depAddr, valAddr)
 	if !found {
 		return sdk.NewCoin(tokens.Denom, sdk.NewInt(0)), time.Time{}, types.ErrNoDepositForAddress
 	}
 
 	depPool, found := k.GetDepositPool(ctx, valAddr)
 	if !found {
-		return sdk.NewCoin(tokens.Denom, sdk.NewInt(0)), time.Time{}, types.ErrNoDepositPoolForValidator
+		panic("found deposit but not the deposit pool")
```

Comment:
Add a comment to inform that when a deposit pool is removed, all deposit associated to it are also removed

Reply:
I agree, with a comment it will be clearer.